### PR TITLE
Add const `MaterialStatePropertyAll` class.

### PR DESCRIPTION
**Pull request by darrenaustin**
Currently if you want to create a `MaterialStateProperty` that always returns the same value regardless of the current states, you could use the `.all` static method:

```dart
final color = MaterialStateColor.all<Color>(Colors.blue);
```

However this can be a const value which makes it harder to use with const ButtonStyles and other structures that could be const otherwise.

This PR adds a new `MaterialStatePropertyAll` class that allows you to convert the above code to:

```dart
const color = MaterialStateColorAll<Color>(Colors.blue);
```

Which can then be used in other const structures.

We may deprecate `MaterialStateProperty.all` in the future, but didn't want to do this without a dart fix for it. As it turns out, this isn't a case that can be [currently handled by dart fix](https://github.com/dart-lang/sdk/issues/49056). When it is, we will probably deprecate it.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

File: packages/flutter/lib/src/material/material_state.dart [hunk 1]
```diff
@@ -651,7 +651,9 @@ abstract class MaterialStateProperty<T> {
 
   /// Convenience method for creating a [MaterialStateProperty] that resolves
   /// to a single value for all states.
-  static MaterialStateProperty<T> all<T>(T value) => _MaterialStatePropertyAll<T>(value);
+  ///
+  /// If you need a const value, use [MaterialStatePropertyAll] directly.
+  static MaterialStateProperty<T> all<T>(T value) => MaterialStatePropertyAll<T>(value);
```
Comment (goderbauer) [inline packages/flutter/lib/src/material/material_state.dart:656]: Add a TODO to deprecate this and link it to the dartfix issue that needs to be resolved to do so?

**Review by goderbauer — APPROVED** [review]
LGTM

**Comment by TahaTesser** [comment]
@darrenaustin 
This is great!

**Comment by darrenaustin** [comment]
> This is great!

Thanks, but all credit to @HansMuller for the idea.

**Pull request merged**